@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -347,9 +347,65 @@ export class TiltBasedClipPlanesEvaluator extends AltitudeBasedClipPlanesEvaluat
             clipPlanes.far = Math.min(clipPlanes.far, clipPlanes.near * this.nearFarMaxRatio);
             return clipPlanes;
         } else {
-            // Fallback to base implemenation case we don't yet have support for camera
-            // tilting when using spherical projections.
-            return super.evaluateClipPlanes(camera, projection);
+            const clipPlanes = { near: 0.0, far: 0.0 };
+
+            // Near plance calculus is pretty straightforward and does not depent on camera tilt:
+            clipPlanes.near = this.getCameraAltitude(camera, projection) - this.nearMargin;
+            // For far plane we need to find tangent to sphere which gives the maximum
+            // visible distance with regards to sphere curvature and camera position.
+            const worldOrigin = new THREE.Vector3(0, 0, 0);
+            // Acquire camera to origin vector.
+            const cameraToOrigin = worldOrigin.sub(camera.position);
+            // Extract camera X, Y, Z orientation axes into tmp vectors array.
+            camera.matrixWorld.extractBasis(
+                this.m_tmpVectors[0],
+                this.m_tmpVectors[1],
+                this.m_tmpVectors[2]
+            );
+            const xaxis = this.m_tmpVectors[0];
+            const yaxis = this.m_tmpVectors[1];
+            const r = EarthConstants.EQUATORIAL_RADIUS + this.farMargin;
+            const d = cameraToOrigin.length();
+            const A = cameraToOrigin.clone().multiplyScalar(1 / d);
```

Comment:
Can you also avoid creating new objects and clone()'s here?

Reply:
Ok, in next update I will do as much as I can, explanations will be in comments.